tests/test_utils.py

```diff
@@ -410,6 +410,29 @@ def test_get_compile_command_with_config(tmpdir_cwd, config_file, expected_comma
         assert get_compile_command(ctx) == expected_command
 
 
+def test_get_compile_command_does_not_include_default_config_if_reqs_file_in_subdir(
+    tmpdir_cwd,
+):
+    """
+    Test that ``get_compile_command`` does not include default config file
+    if requirements file is in a subdirectory.
+    Regression test for issue GH-1903.
+    """
+    default_config_file = Path("pyproject.toml")
+    default_config_file.touch()
```

Comment:
Can we also test the case when the `[tool.pip-tools]` section is present and empty?

Also, how about a `.pip-tools.toml` with an empty or missing `[tool.pip-tools]` and `pyproject.toml` with a non-empty one? I think that the `pyproject.toml` shouldn't be taken into account in this case.

Reply:
Okay, I'll add those combinations to parametrized tests 👍🏻

Reply:
Addressed in 64445d00524f2b76af6825c2f79424b179d914bc

Reply:
I was hoping to see what happens if both files exist too..

Reply:
![](https://c.tenor.com/EYPJjOVJOHYAAAAC/inception-deeper.gif)

Reply:
This looks like out of scope of `get_compile_command`. We can add more unit tests to `select_config_file()`.

Reply:
If it's out of the scope, it'd be good to add the tests separately.

Reply:
Addressed In 783a6e0cb37ed52c2cdf07a88c5a5e5bdfca8d06